oioioi/forum/views.py

```diff
@@ -52,9 +53,21 @@
 @enforce_condition(contest_exists & can_enter_contest)
 @enforce_condition(forum_exists_and_visible & is_proper_forum)
 def forum_view(request):
-    category_set = request.contest.forum.category_set.prefetch_related(
-        'thread_set', 'thread_set__post_set'
-    ).all()
+    categories = []
+    for category in request.contest.forum.category_set.all():
+        threads = category.thread_set.annotate(
+            post_count=Count('post'),
+            reported_count=Count('post', filter=Q(post__reported=True)),
+        ).all()
+        post_count = sum([thread.post_count for thread in threads])
```

Comment:
You are creating a list here just to sum it's all elements. It wastes memory. I suggest to use a [generator expression](https://docs.python.org/3/reference/expressions.html#generator-expressions) like so:
```
sum( (thread.post_count for thread in threads) )
```